src/main/core/Controller/AdministrationController.php

```diff
@@ -40,19 +42,21 @@ class AdministrationController
     /** @var ToolManager */
     private $toolManager;
 
-    /**
-     * AdministrationController constructor.
-     */
+    /** @var StrictDispatcher */
+    private $strictDispatcher;
+
     public function __construct(
         AuthorizationCheckerInterface $authorization,
         TokenStorageInterface $tokenStorage,
         EventDispatcherInterface $eventDispatcher,
-        ToolManager $toolManager
+        ToolManager $toolManager,
+        StrictDispatcher $strictDispatcher
     ) {
         $this->authorization = $authorization;
         $this->eventDispatcher = $eventDispatcher;
         $this->toolManager = $toolManager;
         $this->tokenStorage = $tokenStorage;
+        $this->strictDispatcher = $strictDispatcher;
```

Comment:
You should only grab one dispatcher (the `StrictDispatcher` for now, because it adds some validation on listeners) and replace calls to the standard dispatcher.

Reply:
Done, thx :)